src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java

```diff
@@ -60,15 +109,73 @@ public enum AttackState {
     public int attackDelayHoldoffTicks = ATTACK_DELAY_NONE;
 
     public AttackState attackState = AttackState.NOT_ATTACKING;
+    // The state of the renderer, will lag a few cycles behind the plugin's state
```

Comment:
Why is that?

Reply:
Sorry I should have also been clearer here.

"cycles" in this comment refers to the `client.getGameCycle()` method

``` java
	/**
	 * Gets the local clients game cycle.
	 * <p>
	 * Note: This value is incremented every 20ms.
	 *
	 * @return the game cycle
	 */
	int getGameCycle();
```

So the comment "The state of the renderer, will lag a few cycles behind the plugin's state" is not talking about ticks. The reason for this lag in the rendered compared to the state is that `onGameTick` will occur very early in overall Tick, about 1-5 cycles (out of the 30 cycles in a single tick), so the state is set early here. Then very late at the end of that game tick the render override is called `@Override public Dimension render(Graphics2D graphics)()` - something like the 15-25th cycle in a given tick. Take these numbers with a grain of salt, and I can't explain the ordering because it's somewhere in the runelite guts.

This de-sync is a great heuristic for when it's too late in a given tick for a weapon swap to take effect. As by the time the render is called any weapon swaps will happen on the next tick. But any swaps noticed by `onVarClientIntChanged` which happen before the renderer is insync will be happening right now and affect this tick. Hopefully that explanation helps! Let me know if you want it codified as a comment into the plugin.